torch/distributed/_tensor/examples/advanced_module_tracker.py

```diff
@@ -0,0 +1,39 @@
+import torch
+from torch.autograd.graph import register_multi_grad_hook
+
+from torch.nn.modules.module import (
+    register_module_forward_hook,
+    register_module_forward_pre_hook,
+)
+from torch.utils._pytree import tree_flatten
+
+from torch.utils.module_tracker import ModuleTracker
+
+
+class AdvancedModuleTracker(ModuleTracker):
```

Comment:
I wonder if we can come up with a more descriptive name for this

Reply:
And maybe add a little bit of docstring explains what this is for?